tiledb/api/src/datatype/mod.rs

```diff
@@ -381,6 +386,56 @@ impl Datatype {
             || self.is_datetime_type()
             || self.is_time_type()
     }
+
+    #[cfg(test)]
```

Comment:
Huh, I don't think I've seen this except for a `mod tests` item before, but this makes sense. Neat.

Reply:
I don't remember what crate I was looking through ages ago, but they were using a similar pattern for cross platform code that was so awesome that it just stuck.

```rust
#[cfg(unix)]
fn do_thing(&self) {
    ....
}

#[cfg(windows)]
fn do_thing(&self) {
   ...
}
```

Having just spent a year working on a cross platform C++ library for a living, the simplicity of that made it stick pretty thoroughly.